services/blockchain-app-registry/tests/functional/shared/utils/downloadRepository.test.js

```diff
@@ -0,0 +1,219 @@
+	it('should return event payload when called with a list of changed files', async () => {
+		const changedFiles = [
+			'alphanet/Lisk/nativetokens.json',
+			'betanet/Lisk/nativetokens.json',
+			'devnet/Lisk/nativetokens.json',
+			'Unknown/Lisk/nativetokens.json',
+			'alphanet/Enevti/nativetokens.json',
+		];
+		const response = await buildEventPayload(changedFiles);
+		expect(response).toEqual({
+			alphanet: ['Lisk', 'Enevti'],
+			betanet: ['Lisk'],
+			devnet: ['Lisk'],
+			mainnet: [],
+			testnet: [],
+		});
+	});
+
+	it('should return event payload when called with empty changed files', async () => {
+		const response = await buildEventPayload([]);
+		expect(response).toEqual({
+			alphanet: [],
+			betanet: [],
+			devnet: [],
+			mainnet: [],
+			testnet: [],
+		});
+	});
+
+	it('should return event payload when called with undefined changed files', async () => {
+		const response = await buildEventPayload(undefined);
+		expect(response).toEqual({
+			alphanet: [],
+			betanet: [],
+			devnet: [],
+			mainnet: [],
+			testnet: [],
+		});
+	});
+
+	it('should return event payload when called with null changed files', async () => {
+		const response = await buildEventPayload(null);
+		expect(response).toEqual({
+			alphanet: [],
+			betanet: [],
+			devnet: [],
+			mainnet: [],
+			testnet: [],
+		});
+	});
+});
+
+describe('Test downloadRepositoryToFS method', () => {
+	const enevtiAppJsonFilePath = path.resolve(`${__dirname}/../../data/app-registry/devnet/Enevti/app.json`);
+
+	it('should download repository correctly for first time', async () => {
+		await rmdir(config.dataDir);
+		await downloadRepositoryToFS();
+		expect(await exists(enevtiAppJsonFilePath)).toEqual(true);
+	});
+
+	it('should update repository correctly when repository is already downloaded before', async () => {
+		const lastSyncedCommitHash = 'dc94ddae2aa3a9534a760e9e1c0425b6dcda38e8';
+
+		await rmdir(enevtiAppJsonFilePath);
+		await keyValueTable.set(
+			KV_STORE_KEY.COMMIT_HASH_UNTIL_LAST_SYNC,
+			lastSyncedCommitHash,
+		);
+		await downloadRepositoryToFS();
+		expect(await exists(enevtiAppJsonFilePath)).toEqual(true);
+		await keyValueTable.delete(KV_STORE_KEY.COMMIT_HASH_UNTIL_LAST_SYNC);
+	});
+});
+
+describe('Test syncWithRemoteRepo method', () => {
+	const lastSyncedCommitHash = 'dc94ddae2aa3a9534a760e9e1c0425b6dcda38e8';
+	const enevtiAppJsonFilePath = path.resolve(`${__dirname}/../../data/app-registry/devnet/Enevti/app.json`);
+	beforeAll(async () => {
+		// Set last sync commit hash in db and remove existing file
+		await keyValueTable.set(
+			KV_STORE_KEY.COMMIT_HASH_UNTIL_LAST_SYNC,
+			lastSyncedCommitHash,
+		);
+		await rmdir(enevtiAppJsonFilePath);
+	});
+	afterAll(async () => keyValueTable.delete(KV_STORE_KEY.COMMIT_HASH_UNTIL_LAST_SYNC));
+
+	it('should sync repository upto latest commit', async () => {
+		await syncWithRemoteRepo();
+		expect(await exists(enevtiAppJsonFilePath)).toEqual(true);
```

Comment:
This test is incorrect. You should check the last commit hash instead .

Reply:
I think this is fine. Checking for latest commit hash will never download anything.